src/axom/quest/examples/quest_distributed_distance_query_example.cpp

```diff
@@ -0,0 +1,881 @@
+  ObjectMeshWrapper object_mesh_wrapper(
+    objectDS.getRoot()->createGroup("object_mesh"));
+
+  object_mesh_wrapper.generateCircleMesh(params.circleRadius,
+                                         params.circlePoints);
+  object_mesh_wrapper.saveMesh();
+
+  //---------------------------------------------------------------------------
+  // Load computational mesh and generate a particle mesh over its nodes
+  // These will be used to query the closest points on the object mesh(es)
+  //---------------------------------------------------------------------------
+  QueryMeshWrapper query_mesh_wrapper;
+
+  query_mesh_wrapper.setupMesh(params.getDCMeshName(), params.meshFile);
+  query_mesh_wrapper.printMeshInfo();
+  query_mesh_wrapper.setupParticleMesh();
+
+  // Copy mesh nodes into qpts array
+  auto qPts = query_mesh_wrapper.getVertexPositions<PointArray>();
+  const int nQueryPts = qPts.size();
+
+  //---------------------------------------------------------------------------
+  // Initialize spatial index for querying points, and run query
+  //---------------------------------------------------------------------------
+
+  auto init_str =
+    axom::fmt::format("{:=^80}",
+                      axom::fmt::format("Initializing BVH tree over {} points",
+                                        params.circlePoints));
+
+  auto query_str = axom::fmt::format(
+    "{:=^80}",
+    axom::fmt::format("Computing closest points for {} query points", nQueryPts));
+
+  axom::utilities::Timer initTimer(false);
+  axom::utilities::Timer queryTimer(false);
+
+  // Convert blueprint representation from sidre to conduit
+  conduit::Node object_mesh_node;
+  object_mesh_wrapper.getBlueprintGroup()->createNativeLayout(object_mesh_node);
+
+  conduit::Node query_mesh_node;
+  query_mesh_wrapper.getBlueprintGroup()->createNativeLayout(query_mesh_node);
+
+  // Create distributed closest point query object and set some parameters
+  quest::DistributedClosestPoint query;
+  query.setRuntimePolicy(params.policy);
+  query.setDimension(DIM);
+  query.setVerbosity(params.isVerbose());
+  query.setObjectMesh(object_mesh_node, object_mesh_wrapper.getCoordsetName());
+
+  // Build the spatial index over the object on each rank
+  SLIC_INFO(init_str);
+  initTimer.start();
+  query.generateBVHTree();
+  initTimer.stop();
+
+  // Run the distributed closest point query over the nodes of the computational mesh
+  SLIC_INFO(query_str);
+  queryTimer.start();
+  query.computeClosestPoints(query_mesh_node,
+                             query_mesh_wrapper.getCoordsetName());
+  queryTimer.stop();
+
+  // Output some timing stats
+  auto getMinMax =
+    [](double inVal, double& minVal, double& maxVal, double& sumVal) {
+      MPI_Allreduce(&inVal, &minVal, 1, MPI_DOUBLE, MPI_MIN, MPI_COMM_WORLD);
+      MPI_Allreduce(&inVal, &maxVal, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD);
+      MPI_Allreduce(&inVal, &sumVal, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+    };
+  {
```

Comment:
Why do you introduce a scope here?  And if a scope is desirable here, perhaps to group variables that aren't elsewhere used, why not put `getMinMax` inside it?

Reply:
The scope was just to make the logging block a bit cleaner. 
I'll add a comment at the top to make it a bit clearer.

The `getMinMax` is used in a different scope below, so cannot be in this one.